### PR TITLE
Response/decline throttle

### PR DESCRIPTION
**Pull request by diegoserranoa**
Updates the SDK to accept resurvey_throttle and decline_resurvey_throttle.
In case the end_user responds to the survey, the value of resurvey_throttle
will be used to set the expiration date of the surveyedAt value of
SharedPreferences (cookie), resurvey_throttle default value is 90.
In case the end_user declines the survey, the value of decline_resurvey_throttle
will be used as the expiration date, default value is 30.

Other fixes:
-Catch NullPointerException for Activity that cause crash
-Update tests
-Add tests for PreferencesUtils
-Update version of Gradle
-Add log for last_seen and surveyed 'cookie' values
-Clean code using Constants.TAG

To test this manually, you can complete a survey, then run the app again, it should log a message: "surveyed(expiration days 90): 0d ago" (90 being the default value for responses). 

To erase the "cookies", delete the app's data. Depending on which version of Android you're using, you can do this by long pressing the app's icon, a "Data Info" should appear on the top of the screen, drag the icon here. Now select "Clear Data". 

Now change the end_user value and try declining the survey. When running the app it should say "surveyed(expiration days 30): 0d ago" (30 being the default value for declines).

Trello: https://trello.com/c/zBEZzmCW/732-1-custom-throttle-for-decline-response-android-update-the-clients-need-the-new-response-resurvey-date-and-decline-resurvey-date-

### CODE REVIEW

File: androidsdk/src/main/java/com/wootric/androidsdk/Constants.java [hunk 1]
```diff
@@ -30,4 +30,8 @@
     public static final long DAY_IN_MILLIS = 1000L *60L *60L *24L;
 
     public static final int NOT_SET = -1;
+
+    public static final long DEFAULT_RESURVEY_DAYS = 90L;
```
Comment (prabhatjha) [inline androidsdk/src/main/java/com/wootric/androidsdk/Constants.java]: Does this work similar to how web beacon works in terms of cookie expiry? So if resurvey_throttle is set to 90 days, SDK would not event hit our eligibility servers for 90 days.

Comment (diegoserranoa) [inline androidsdk/src/main/java/com/wootric/androidsdk/Constants.java]: @prabhatjha I checked the web beacon to be sure. On web beacon, if cookie `surveyed` exists, then it doesn't hit eligibility server. On code, that cookie's expiration date is supposed to be set with the `resurvey_throttle` value. However, that value is not being sent right now, which causes the cookie to expire when browsing session ends. Without that cookie, web beacon always hits eligibility server.

**Comment by prabhatjha** [comment]
How did you test it in app/emulator?

**Comment by prabhatjha** [comment]
:shipit:

**Pull request merged**